src/main/java/com/atlassian/db/replica/api/AuroraMultiReplicaConsistency.java

```diff
@@ -26,7 +26,7 @@ public void write(Connection main) {
     @Override
     public boolean isConsistent(Supplier<Connection> replicaSupplier) {
         return cluster.getReplicas(replicaSupplier).stream()
-            .allMatch(replica -> replicaConsistency.isConsistent(() -> replica.getConnectionSupplier().get()));
+            .allMatch(replica -> replicaConsistency.isConsistent(replica.getConnectionSupplier()));
```

Comment:
This place is still leaking the connection right? Closing connection by ReplicaConsistency is not part of the contract

Reply:
IMO if someone takes connection from supplier then it's in their responsibility to correctly call close operation.

Reply:
Maybe, maybe not. The point is it's still leaking the connections.
We should either close it here or make it explicit in Java docs that not closing the connections from the supplier will lead to connection leaks.

I'm fine with moving it to separate PR to iterate faster. :)

Reply:
> in Java docs that not closing the connections from the supplier will lead to connection leaks.

like it